### PR TITLE
include/byteorder: drop init_le*() helpers

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
@athanatos i found that we are using helpers like `init_le32()` everywhere in the place of the constructor of `ceph_le32()`, so that's why i want to drop it.

will rebase this change after #39828 gets merged.

File: src/include/byteorder.h [hunk 1]
```diff
@@ -16,6 +16,10 @@ struct ceph_le {
 private:
   T v;
 public:
+  ceph_le() = default;
```
Comment (tchaikov) [inline src/include/byteorder.h:19]: will replace `ceph_le` as a template alias of `endian_arithmetic`, once we have the luxury of using boost with the https://github.com/boostorg/endian/issues/28 fix.

File: src/test/librados/c_read_operations.cc [hunk 2]
```diff
@@ -407,7 +407,7 @@ TEST_F(CReadOpsTest, Checksum) {
   }
 
   {
-    ceph_le32 init_value = init_le32(-1);
+    ceph_le32 init_value = ceph_le32(-1);
```
Comment (ronen-fr) [inline src/test/librados/c_read_operations.cc:410]: we should probably stop assigning negative numbers to unsigned. Maybe use UINT*_MAX?

**Review by ronen-fr — APPROVED** [review]
LGTM (and see one comment)

File: src/test/librados/c_read_operations.cc (hunk 2 above)
Comment (tchaikov) [reply src/test/librados/c_read_operations.cc]: will address this in a follow-up PR.

**Review by cbodley — APPROVED** [review]

**Review by dillaman — APPROVED** [review]
RBD lgtm -- thx

**Review by athanatos — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-03-11_07:16:51-rados-wip-kefu-testing-2021-03-11-1238-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-03-11_16:25:48-rados-wip-kefu-testing-2021-03-11-2158-distro-basic-gibba/
- https://pulpito.ceph.com/kchai-2021-03-12_05:42:44-rados-wip-kefu-testing-2021-03-12-1106-distro-basic-smithi/

**Pull request merged**